.github/workflows/code-formating-check.yml

```diff
@@ -0,0 +1,29 @@
+name: "Check Formating"
+on: [push, pull_request]
+
+jobs:
+  check_code_formatting:
+    name: Check code formatting with Black
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1.1.1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Restore cache for check_code_formatting
```

Comment:
the cacher here is strange as later you install specific black version... so once it is installed and cached, even you change the version it will be loaded the cached version as you also do not force for upgrade